### PR TITLE
exit after panic

### PR DESCRIPTION
**Pull request by real-danm**
due to state and locking issues in the sfu, just let panics crash.

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]
Leaving `Recover` in utils would be good (actually needed if we want to catch and log). We probably would lose the participant logger if the catch is at top level, but that is okay I think.

Summarising, leave utils functions in there (I think it is okay to remove `RecoverSilent` as that should not be used) and remove the call sites.

File: pkg/rtc/utils.go [hunk 1]
```diff
@@ -126,7 +123,9 @@ func Recover(l logger.Logger) {
 		default:
 			err = errors.New("unknown panic")
 		}
-		l.Errorw("recovered panic", err, "panic", r)
+		l.Errorw("Panic", err, "panic", r)
```
Comment (boks1971) [inline pkg/rtc/utils.go:126]: use lower case `panic`, we use lower case log lines, probably some places have upper case, but we should standardize

File: pkg/rtc/utils.go [hunk 2]
```diff
@@ -126,7 +123,9 @@ func Recover(l logger.Logger) {
...
+
+		os.Exit(1)
```
Comment (boks1971) [inline pkg/rtc/utils.go:128]: I think this should not be here. It should be in `main`. Callers can choose to let things `recover` or exit. As a utility, this is just doing `recover`.

**Review by matkam — APPROVED** [review]
LGTM

**Review by paulwe — APPROVED** [review]

File: pkg/rtc/utils.go [hunk 3]
```diff
@@ -108,15 +108,12 @@ func IsEOF(err error) bool {
 	return err == io.ErrClosedPipe || err == io.EOF
 }
 
-func RecoverSilent() {
-	recover()
-}
-
-func Recover(l logger.Logger) {
+func Recover(l logger.Logger) any {
```
Comment (boks1971) [inline pkg/rtc/utils.go:111]: what `any` is an accepted type in golang? :TIL:

**Review by boks1971 — APPROVED** [review]
Looks good @real-danm . Some more clean up would good I think. But, don't want to delay this further. Sorry, have been dragging on this. Other things I was thinking was
1. Make a utility function called `MaybeRecoverAndExit()` which can called from all places. Saves a few lines (all the recover defers become single line). That function wraps calling recover and os.Exit.
2. The `Recover` function I think is better suited in `pkg/utils` rather than `pkg/rtc`

But, we can make these changes later.

File: pkg/rtc/utils.go (hunk 3 above)
Comment (matkam) [reply pkg/rtc/utils.go:111]: nit: how about returning the error (below) instead. returning `any` isn't as useful.

Comment (real-danm) [reply pkg/rtc/utils.go:111]: the recover builtin returns any, and I am returning that value.

**Pull request merged**